sdk/src/main/scala/com/horizen/SidechainNodeViewHolder.scala

```diff
@@ -299,27 +338,164 @@ class SidechainNodeViewHolder(sidechainSettings: SidechainSettings,
   protected def processLocallyGeneratedTransaction: Receive = {
     case newTxs: LocallyGeneratedTransaction[SidechainTypes#SCBT] =>
       newTxs.txs.foreach(tx => {
-        if (tx.fee() > maxTxFee)
+        if (isReindexing()) {
+          context.system.eventStream.publish(FailedTransaction(tx.asInstanceOf[Transaction].id, new IllegalArgumentException(s"Reindex in progress, unable to process Transaction ${tx.id()}"),
+            immediateFailure = true))
+        } else if(tx.fee() > maxTxFee) {
           context.system.eventStream.publish(FailedTransaction(tx.asInstanceOf[Transaction].id, new IllegalArgumentException(s"Transaction ${tx.id()} with fee of ${tx.fee()} exceed the predefined MaxFee of ${maxTxFee}"),
             immediateFailure = true))
-        else
+        } else {
           txModify(tx)
-
+        }
       })
   }
 
+
+  /**
+   * Process reindex messages
+   */
+  protected def processReindex: Receive = {
+    case SidechainNodeViewHolder.ReceivableMessages.ReindexStep(proceedIfNeeded: Boolean) => {
+
+      reindexStep(proceedIfNeeded) match {
+        case Success(proceed) => {
+          if (proceed) {
+            self ! SidechainNodeViewHolder.ReceivableMessages.ReindexStep(proceedIfNeeded)
+          }
+        }
```

Comment:
proceedIfNeeded is used for  debug pourpuses to execute just only one step.
it is passed below to the reindexStep method because there is a "special" case of step 0 that must be always followed by step 1 regardless of this flag. so the inner method logic will:
- always return proceed = true after step 0
- return the same input flag for every other step